src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebHostEnvironmentTests.cs

```diff
@@ -58,16 +60,94 @@ public void ApplyEnvironmentSettingsUsesTheCorrectKeysAndProperties()
+            Assert.IsType<PhysicalFileProvider>(environment.WebRootFileProvider);
+
+            Assert.Equal(tempPath, ((PhysicalFileProvider)environment.ContentRootFileProvider).Root);
+            Assert.Equal(tempPath, ((PhysicalFileProvider)environment.WebRootFileProvider).Root);
+        }
+
+        [Fact]
+        public void RelativePathsAreMappedToFullPaths()
+        {
+            var environment = new WebHostEnvironment();
+            var relativeRootPath = "some-relative-path";
+            var relativeSubPath = "some-other-relative-path";
+            var fullContentRoot = Path.Combine(AppContext.BaseDirectory, relativeRootPath);
+            
+            // ContentRootPath is mapped relative to AppContext.BaseDirectory
+            environment.ContentRootPath = relativeRootPath;
+            Assert.Equal(fullContentRoot, environment.ContentRootPath);
+
+            // WebRootPath is mapped relative to ContentRootPath
+            environment.WebRootPath = relativeSubPath;
+            Assert.Equal(Path.Combine(fullContentRoot, relativeSubPath), environment.WebRootPath);
+        }
+
+        [Fact]
+        public void UnsettingPathsFallsBackToDefaults()
+        {
+            var environment = new WebHostEnvironment();
+            var defaultWebRootPath = Path.Combine(environment.ContentRootPath, "wwwroot");
+            var webRootPath = Path.GetTempPath();
+
+            environment.WebRootPath = webRootPath;
+
+            Assert.Equal(webRootPath, environment.WebRootPath);
+            Assert.Equal(webRootPath, ((PhysicalFileProvider)environment.WebRootFileProvider).Root);
+
+            // Setting WebRootPath to fallsback to default
+            environment.WebRootPath = null;
+            Assert.Equal(defaultWebRootPath, environment.WebRootPath);
+
+            // Setting ContentRootPath to null falls back to current directory
+            environment.ContentRootPath = null;
+            Assert.Equal(AppContext.BaseDirectory, environment.ContentRootPath);
+            Assert.Equal(AppContext.BaseDirectory, ((PhysicalFileProvider)environment.ContentRootFileProvider).Root);
+        }
+
+        [Fact]
+        public void SetContentRootAfterRelativeWebRoot()
+        {
+            var environment = new WebHostEnvironment();
+            var webRootPath = "some-relative-path";
+            var tempPath = Path.GetTempPath();
+
+            environment.WebRootPath = webRootPath;
+
+            Assert.Equal(Path.Combine(AppContext.BaseDirectory, webRootPath), environment.WebRootPath);
+            Assert.Equal(Directory.GetCurrentDirectory(), environment.ContentRootPath);
+
+            // Setting the ContentRootPath after setting a relative WebRootPath
+            environment.ContentRootPath = tempPath;
+
+            Assert.Equal(tempPath, environment.ContentRootPath);
+            Assert.Equal(Path.Combine(tempPath, webRootPath), environment.WebRootPath);
```

Comment:
Can we verify that the PhysicalFileProviders have the correct `Root` in this test too?